pycfmodel/model/resources/resource.py

```diff
@@ -43,4 +45,28 @@ def has_hardcoded_credentials(self) -> bool:
 
     @property
     def policy_documents(self) -> List[OptionallyNamedPolicyDocument]:
-        return []
+        """
+        Returns a list with all the optionally named policy documents in this resource within its properties.
+        Every resource has a Properties field, if not, it's a malformed CloudFormation template.
+        """
+        policy_documents = []
+        self.obtain_policy_documents(
+            policy_documents=policy_documents, properties=list(self.Properties.__dict__.values())
+        )
+        return policy_documents
+
+    def obtain_policy_documents(self, policy_documents: List[OptionallyNamedPolicyDocument], properties: List[Any]):
```

Comment:
```suggestion
    def obtain_policy_documents(self, policy_documents: List[Properties], properties: List[Any]):
```
It can not be assured to only pass OptionallyNamedPolicyDocument

Reply:
Right! But it cannot be assured to be a list of properties neither, adapted it to be just a list in 323fc4c